logits_processor_zoo/transformers/base.py

```diff
@@ -0,0 +1,43 @@
+#
+# SPDX-FileCopyrightText: Copyright (c) 1993-2024 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import torch
+
+
+class BaseLogitsProcessor:
+    def __init__(self):
+        self.input_len = None
+
+    def _reset(self):
+        pass
+
+    def _reset_if_new_batch(self, input_ids: torch.LongTensor):
```

Comment:
Could you add a docstring explaining this logic?

Reply:
I didn't add a docstring because it is a private function that I dont want to expose to users but I added comments to make it more clear if it is okay?

Reply:
Feel free to add a comment, not curcial.